### PR TITLE
fix(iam): failed to batch import user with roles

### LINKED ISSUE #1900: [Bug]: NPE occurs when batch import user
### ODC version

4.2.4

### OB version

-

### What happened?

When batch import user from xlsx file and set roles, there will be an NPE . 
image

image

### What did you expect to happen?

import should succeed

### How can we reproduce it (as minimally and precisely as possible)?

.

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:

In `UserBatchImportPreviewer` , we used `Collections.emptyList()` incorrectly. It caused failures to import user.
Also optimized the error message of verification failure.

#### Which issue(s) this PR fixes:

Fixes #1900

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/iam/UserBatchImportPreviewer.java [hunk 1]
```diff
@@ -59,21 +60,12 @@
...
-    private static final Pattern REG_LOWERCASE = Pattern.compile(".*[a-z]+.*");
-    private static final Pattern REG_SYMBOL = Pattern.compile(".*[._+@#$%]+.*");
-    private static final Pattern SPACE_PATTERN = Pattern.compile("^(?=\\S).+(?<=[^.\\s])$");
-    public static final int PASSWORD_MAX_LENGTH = 32;
-    public static final int PASSWORD_MIN_LENGTH = 8;
-    public static final int NAME_MAX_LENGTH = 64;
-    public static final int ACCOUNTNAME_MAX_LENGTH = 64;
-    public static final int ACCOUNTNAME_MIN_LENGTH = 4;
+    private static final Pattern ACCOUNT_NAME_PATTERN = Pattern.compile("^[\\w.+@#$%]{4,64}$");
+    private static final Pattern PASSWORD_PATTERN = Pattern.compile(
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/iam/UserBatchImportPreviewer.java:64]: Wow! How can you write so complex Regex expression? 👍🏻

File: server/odc-service/src/main/java/com/oceanbase/odc/service/iam/UserBatchImportPreviewer.java [hunk 2]
```diff
@@ -120,33 +118,33 @@ private boolean checkFileType(MultipartFile file) {
...
-        }
+        Verify.verify(SPACE_PATTERN.matcher(name).matches(),
+                "username length should not exceed 64 and not start or end with spaces");
+        batchImportUser.setName(name);
+
+        String accountName = map.get(USER_ACCOUNTNAME.getLocalizedMessage());
+        Verify.verify(!accountNames.contains(accountName), "account name already exists");
+        Verify.verify(ACCOUNT_NAME_PATTERN.matcher(accountName).matches(),
+                "account name must be 4 to 64 characters long "
+                        + "and consist of letters, numbers, and special characters in ._+@#$% only");
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/iam/UserBatchImportPreviewer.java:131]: --> "Contains at least two lowercase letters, two uppercase letters, two numbers, and two special characters."

**Review by smallsheeeep — APPROVED** [review]
LGTM

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**